### ISSUE TITLE
Fix broken Docker build

### ISSUE THREAD

**Issue by Jarmos-san** [Kampfkarren/selene#550]
This issue intends to track the discussions to fix the broken Docker build. Once the Docker build is fixed #541 can possibly be merged.

On top of it, the current `Dockerfile` is configured to use and older version of the Rust Docker Image. Ideally, the latest image should be used ([see comment](https://github.com/Kampfkarren/selene/pull/541#discussion_r1306611850)) or be able to pass a specific version Rust version using [build-time variables](https://docs.docker.com/engine/reference/commandline/build/#build-arg). In other words, the Docker image could be built by running the following commands:

```console
docker build --build-arg RUST_VERSION="1.72-slim-buster" .
```
Running the command above would instruct Docker to use the `1.72-slim-buster` version instead of the default `1-alpine` version as stated in the `Dockerfile` below:

```Dockerfile
# "1-alpine" is the latest Alpine version of the latest version of Rust
ARG RUST_VERSION="1-alpine"

# Utilise multi-stage builds to keep the final image small and efficient
FROM rust:${RUST_VERSION} AS builder

# ... rest of the contents of the Dockerfile
```

In addition to the source code fixes, the documentations needs to be updated as well for posterity and proper usage of the tools.

**Comment by Jarmos-san** [comment]
@Kampfkarren you can link this issue with #549 so that it is automatically closed once that PR is merged?

**Comment by Kampfkarren** [comment]
You could do that with "Closes #550" in your issue 😄
